src/DotLiquid/Util/R.cs

```diff
@@ -59,6 +59,19 @@ public static List<string> Scan(string input, Regex regex)
                 .ToList();
         }
 
+        /// <summary>
+        /// Overload that only works when the pattern contains two groups. The callback
+        /// is called for each match, passing the two group values.
+        /// </summary>
+        /// <param name="input"></param>
+        /// <param name="regex"></param>
+        /// <param name="callback"></param>
+        public static void Scan(string input, Regex regex, Action<string, string> callback)
+        {
+            foreach (Match match in regex.Matches(input))
+                callback(match.Groups[1].Value, match.Groups[2].Value);

```

Comment:
Do we really need this, or can we inline this to the 3 callers? Seems easy enough to do. And if yes, I question the public part (which is why we need to obsolete and not delete the old method)

Reply:
As a general rule, I think if it's used in at least three places, it's a good candidate for refactor/function. So I'd keep it.

Reply:
Add 5 lines of code to reduce 3 places from 2 lines to 1. Net total of 2 extra lines. I disagree, especially since it isn't more readable (what is action<string, string>?) and there is overhead of n+1 stacktrace changes depending on how many matches are found.

Reply:
No worries. I've removed it, but still marked the old one Obsolete.